odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt

```diff
@@ -188,4 +200,115 @@ class ODKFormsHandler @Inject constructor(
         }
         return document
     }
+
+
+    override fun openForm(formId: String, context: Context) {
+        CoroutineScope(Job()).launch {
+            // Delete any saved instances of this form
+            val savedInstances = instancesRepository.getAllByFormId(formId)
+            for (instance in savedInstances) {
+                if (instance.status == Instance.STATUS_INCOMPLETE) {
+                    instancesRepository.delete(instance.dbId)
+                }
+            }
+            val requiredForm = formsDatabaseInteractor.getLatestFormById(formId)
+            if (requiredForm == null) {
+                downloadAndOpenForm(formId, context)
+            }
+            else {
+                val xmlFile = File(requiredForm.formFilePath)
+                if (xmlFile.exists() && (requiredForm.formMediaPath == null || mediaExists(requiredForm))) {
+                    formsInteractor.openFormWithFormId(formId, context)
```

Comment:
you do not need to use `formsInteractor` here. You're already calling `openFormWithForId` inside an implementation of `FormsInteractor`.